internal/utils/kruize/kruize_api.go

```diff
@@ -178,15 +178,96 @@ func Update_recommendations(experiment_name string, interval_end_time time.Time)
 
 }
 
-func Is_valid_recommendation(d []kruizePayload.ListRecommendations) bool {
-	if len(d) > 0 {
-		notifications := d[0].Kubernetes_objects[0].Containers[0].Recommendations.Notifications
-		// 112101 is notification code for "Duration Based Recommendations Available".
-		if _, ok := notifications["112101"]; ok {
-			return true
-		} else {
+func Is_valid_recommendation(recommendation kruizePayload.Recommendation, experiment_name string, maxEndTime time.Time) bool {
+
+	validRecommendationCode := "111000"
+	_, recommendationIsValid := recommendation.Notifications[validRecommendationCode]
+	if recommendationIsValid {
+		// Convert the time object to the expected format
+		formattedMaxEndTime := maxEndTime.UTC().Format("2006-01-02T15:04:05.000Z")
+		recommendationData, timeStampisValid := recommendation.Data[formattedMaxEndTime]
+		if !timeStampisValid {
+			log.Error("recommendation not found for endtime: ", formattedMaxEndTime)
+			invalidRecommendation.Inc()
 			return false
 		}
+		LogKruizeErrors(recommendationData, formattedMaxEndTime, experiment_name)
```

Comment:
If it is discussed internally with the Kruize team regarding not having these error logs on ROS side then we can remove this. Apart from it everything looks good to me 👍 
I have tested PR locally and working as expected.